lib/storage/s3.js

```diff
@@ -1,142 +1,144 @@
 "use strict";
 
-module.exports = function (awsSdk, promise) {
-    class S3 {
-        constructor(config) {
-            this.config = {};
-            this.configure(config);
-            this.awsS3 = null;
-        }
+/**
+ * Handles the functionality between OpenToken.io and
+ * Amazon S3. We explicitly say how we are communicating
+ * and what functions are open to use.
+ *
+ * All methods are asychronous.
+ */
+module.exports = function (awsSdk, config, promise) {
+    var awsS3, s3Bucket;
+
+    awsS3 = null;
+    s3Bucket = null;
+    configure(config.storage);
+
+    /**
+     * Configures options for AWS.
+     *
+     * @param {Object} config
```

Comment:
Would *greatly* prefer that the config passed in uses a namespaced config.  It makes it much easier to switch between multiple configs.  Here's a sample:

```
config = {
    storage: {
        engine: "s3",
        s3: {
            region: "us-east-1"
        }.
        disk: {
            base: "/var/lib/opentoken/"
        }
    }
}
```